src/Home/Prodcuts.tsx

```diff
@@ -2,28 +2,30 @@ import { useContext } from "react";
 import { Link } from "react-router-dom";
 import { CartContext } from "../ProductManagement/CartStore";
 
-import { BaseProduct as Product } from "../ProductManagement/ProductModels";
+import { BaseProduct } from "../ProductManagement/ProductModels";
+import { addToCart } from "../ProductManagement/Cart/AddCart";
+import { UserContext } from "./userStore";
+import { Toaster } from "react-hot-toast";
 
 interface ProductProps {
-  products: Product[];
+  products: BaseProduct[];
 }
 
-const Product = ({ products }: ProductProps) => {
-  const { cartDispatch, cartState } = useContext(CartContext);
+const Products = ({ products }: ProductProps) => {
+  const { cartDispatch } = useContext(CartContext);
+  const { token, username } = useContext(UserContext);
 
-  const addToCart = (_id: string) => {
-    const product: Product | undefined = products.find(
+  const addCart = (_id: string) => {
+    const product: BaseProduct | undefined = products.find(
       (product) => product._id === _id
     );
-    if (product)
-      cartDispatch({
-        type: "ADD_ITEM",
-        payload: { ...product, quantity: 1 },
-      });
-    console.log(cartState);
+    if (product) {
+      addToCart({ product, token, username, cartDispatch });
+    }
```

Comment:
### if token does not exists redirect to login the user add
>create function to redirect to login page if token does not exist when user try to add item to cart 

#4 #3  @20pa5a1210